llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -2325,8 +2325,10 @@ bool AsmPrinter::doFinalization(Module &M) {
     // Emit linkage for the function entry point.
     emitLinkage(&F, FnEntryPointSym);
 
-    // Emit linkage for the function descriptor.
-    emitLinkage(&F, Name);
+    // If address is taken from an extern function, we need to emit linkage for
+    // its function descriptor symbol.
```

Comment:
nit: we can change the comment a little bit here like "if function's address is taken, means it may be called via a function pointer, we need the function descriptor for it."